rhino/src/main/java/org/mozilla/javascript/ScriptRuntime.java

```diff
@@ -280,6 +277,9 @@ public static ScriptableObject initStandardObjects(
             Context cx, ScriptableObject scope, boolean sealed) {
         ScriptableObject s = initSafeStandardObjects(cx, scope, sealed);
 
+        NativeJavaObject.init(s, sealed);
+        NativeJavaMap.init(s, sealed);
```

Comment:
Yes, they should be definitively considered as unsafe. (The initialization will be later moved to liveconnect)